express/dataset.py

```diff
@@ -0,0 +1,212 @@
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
+        return self.manifest.project_name
+
+    def _load_subset(self, name: str) -> dd.DataFrame:
+        # get subset from the manifest
+        subset = self.manifest.get_subset(name)
+        # get its location and fields
+        # TODO remove gcp prefix
+        location = "gcs://" + subset.location + ".parquet"
+        fields = list(subset.fields.keys())
+
+        df = dd.read_parquet(
+            location, columns=fields, storage_options={"project": self.project_name}
+        )
+
+        return df
+
+    def load_data(self, component_spec: ExpressComponent) -> dd.DataFrame:
+        subsets = []
+        for name in component_spec.input_subsets.keys():
+            subset_data = self._load_subset(name)
+            subsets.append(subset_data)
+
+        # TODO this method should return a single dataframe with column_names called subset_field
+        # TODO add index
+        # dataset = concatenate_datasets(subsets)
+
+        # return dataset
+
+    def _upload_index(self, df) -> Index:
+        # get location
+        # TODO remove GCP prefix
+        remote_path = "gcs://" + self.manifest.index.location + ".parquet"
+        # upload to the cloud
+        dd.to_parquet(
+            df,
+            remote_path,
+            storage_options={"project": self.gcs_project},
+            overwrite=True,
+        )
+
+    def add_index(self, output_dataset):
```

Comment:
docstrings and typehints

Reply:
Let's add docstrings in a follow-up PR, have added typehints